src/dodal/devices/i24/pmac.py

```diff
@@ -150,17 +154,19 @@ async def kickoff(self):
         )
 
     @AsyncStatus.wrap
-    async def complete(self):
-        """Stop collecting when the scan status PV goes to 0.
-
-        Args:
-            complete_time (float): total time required by the collection to \
-            finish correctly.
+    async def complete(self, counter_timeout: float = 30):
+        """Stop collecting when the scan status PV goes to 0 or when counter PV hasn't \
+            updated for 30 seconds.
         """
-        scan_complete_time = await self._collection_time_ref().get_value()
-        await wait_for_value(
-            self._status_ref(), ScanState.DONE, timeout=scan_complete_time
-        )
+        try:
+            async for signal, value in observe_signals_value(
+                self._status_ref(), self._counter_ref(), timeout=counter_timeout
+            ):
+                if signal is self._status_ref():
+                    if value == ScanState.DONE:
+                        break
+        except TE:
+            pass
```

Comment:
Must: Why are we swallowing this error? The issue was to make sure it does time out